### PR TITLE
switch to nx-release for publishing

### PR DESCRIPTION
**Pull request by llwt**
This sets up `nx release` for use in dev publishing. We might need some tweaks when it comes time to do the next real publish, but it's been working well so far for me in dev.

To test:
```bash
# in one terminal
nx local-registry

# in a second terminal
nx release-dev nx-react-workshop
```

Builds on #32

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -24,12 +24,18 @@
     {
       "files": ["*.ts", "*.tsx"],
       "extends": ["plugin:@nx/typescript"],
-      "rules": {}
+      "rules": {
+        "@typescript-eslint/no-extra-semi": "error",
```
Comment (meeroslav) [inline .eslintrc.json]: Why was this needed? The ts version of the no extra semi is a fairly expensive rule

Comment (llwt) [reply .eslintrc.json]: I have no idea, I would suspect it's not. This was done by the migrations when updating to the latest Nx. 

I'll revert.

Comment (llwt) [reply .eslintrc.json]: Done.

File: libs/nx-react-workshop/project.json [hunk 2]
```diff
@@ -54,14 +54,8 @@
...
-    "publish": {
-      "executor": "nx:run-commands",
-      "options": {
-        "commands": [
-          "yarn nx build nx-react-workshop",
-          "yarn publish dist/libs/nx-react-workshop"
-        ]
-      }
+    "release-dev": {
+      "command": "nx release version prerelease && nx release publish"
```
Comment (llwt) [inline libs/nx-react-workshop/project.json:58]: See if `nx release canary` works and skips tagging

**Review by meeroslav — APPROVED** [review]
